### PR TITLE
load tile style from uri

### PR DESCRIPTION
**Pull request by ggetz**
If string is provided to `Cesium3DTileStyle`, load tile style from uri.

I added tests in `Cesium3DTileStyle` and `Cesium3DTileset` to check if the style is actually applied.

### CODE REVIEW

File: Source/Scene/Cesium3DTileStyle.js [hunk 1]
```diff
@@ -21,16 +29,44 @@ define([
...
+            }).otherwise(function(error) {
+                that._readyPromise.reject(error);
+            });
+        } else {
+            setup(that, tileset, data);
+            that._readyPromise.resolve(that);
+        }
+    }
+
+    function setup (that, tileset, styleJson) {
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyle.js]: In this function, rename `that` to `style`.

File: Specs/Scene/Cesium3DTileStyleSpec.js [hunk 2]
```diff
@@ -37,10 +74,14 @@ defineSuite([
...
         });
-        expect(style.color).toEqual(new Expression(styleEngine, 'color("#ffffff")'));
 
-        style = new Cesium3DTileStyle(tileset);
-        expect(style.color).toEqual(new Expression(styleEngine, 'color("#ffffff")'));
+        style.readyPromise.then(function(s) {
+            expect(s.color).toEqual(new Expression(styleEngine, 'color("#ffffff")'));
+        });
+
+        //
```
Comment (pjcozzi) [inline Specs/Scene/Cesium3DTileStyleSpec.js]: Do you need this commented out code?

File: Specs/Scene/Cesium3DTilesetSpec.js [hunk 3]
```diff
@@ -1016,4 +1018,25 @@ defineSuite([
...
+
+    it('loads style from uri', function() {
+        // One building in each data set is always located in the center, so point the camera there
+        var center = Cartesian3.fromRadians(centerLongitude, centerLatitude, 5.0);
+        scene.camera.lookAt(center, new HeadingPitchRange(0.0, -1.57, 10.0));
+
+        return Cesium3DTilesTester.loadTileset(scene, withBatchTableUrl).then(function(tileset) {
+            // ${id} < 100 will always evaluate to true
+            var loadedStyle = new Cesium3DTileStyle(tileset, styleUrl);
+            return loadedStyle.readyPromise.then(function(style) {
```
Comment (pjcozzi) [inline Specs/Scene/Cesium3DTilesetSpec.js]: The user shouldn't have to wait for ready.  Instead, the style engine should check if the style is ready before applying it.  If the style is not ready and its apply is called, it should throw a `DeveloperError` (see Model.js for examples).

So this should be as simple as
```javascript
tileset.style = new Cesium3DTileStyle(tileset, styleUrl);
```
Then the style is basically asynchronously applied.

**Comment by pjcozzi** [comment]
Just those comments.

**Comment by ggetz** [comment]
I don't think all these tests should be failing on Travis- They are not failing on my machine.

**Comment by ggetz** [comment]
Updated.

**Comment by pjcozzi** [comment]
Don't worry about travis for this PR.  I need to look at it, in general, for the 3d-tiles branch.

File: Source/Scene/Cesium3DTileStyleEngine.js [hunk 4]
```diff
@@ -113,6 +115,12 @@ define([
             return;
         }
 
+        //>>includeStart('debug', pragmas.debug);
+        if (!style.ready) {
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyleEngine.js]: Move this check to inside `evaluateColor` and `evaluate`.  That is the object that isn't ready, not this.

Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyleEngine.js]: Then add a check for `style.ready` and quick return in `applyStyle` above just like it checks `tileset.ready`.

**Comment by pjcozzi** [comment]
Just those comments.

**Comment by ggetz** [comment]
> Move this check to inside evaluateColor and evaluate. That is the object that isn't ready, not this.

Move this check into `Expression.evaluate` and `Expression.evaluateColor`?

**Comment by ggetz** [comment]
Because if the style is not yet ready, `color` and `show` are not yet set to expressions, so the `evaluate` functions will not be defined.

**Comment by ggetz** [comment]
Updated.

File: Source/Scene/Cesium3DTileStyle.js [hunk 5]
```diff
@@ -53,11 +78,64 @@ define([
...
+        },
+
+        /**
+         * DOC_TBA
+         */
+        show : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
+                if (!this._ready) {
+                    throw new DeveloperError('The style is not loaded.  Use Cesium3DTileStyle.readyPromise or wait for Cesium3DTileStyle.ready to be true.');
```
Comment (pjcozzi) [inline Source/Scene/Cesium3DTileStyle.js]: Did you add tests for these new exceptions?

**Comment by ggetz** [comment]
Updated

**Comment by pjcozzi** [comment]
:+1:

**Pull request merged**